app/models/manageiq/providers/inventory/persister/builder.rb

```diff
@@ -228,7 +229,16 @@ def auto_model_class
         # return the base class.
         #
         # safe_constantize can return different similar class ( some Rails auto-magic :/ )
-        provider_class.to_s == class_name ? provider_class : "::#{name.to_s.classify}".safe_constantize
+        return provider_class if provider_class.to_s == class_name
+
+        klass = "::#{name.to_s.classify}".safe_constantize
+        raise NotSubclassedError, "#{name} should be subclassed under #{manager_class}" if sti?(klass) && !@options[:without_sti]
+
+        klass
+      end
+
+      def sti?(klass)
+        klass.columns_hash.key?("type")
       end
```

Comment:
My mind is blown that this isn't just a built-in public method on a model: https://github.com/rails/rails/blob/6-0-stable/activerecord/lib/active_record/inheritance.rb.  That being said, it's more accurate to do:

```suggestion
      def sti?(klass)
        klass.columns_hash.key?(klass.inheritance_column.to_s)
      end
```